common/http/http.cpp

```diff
@@ -423,5 +425,39 @@ ExponentialBackoff::ExpectedInterval ExponentialBackoff::NextInterval() {
 	return current_interval;
 }
 
+bool HostNameMatchesNoProxy(const string &host, const string &no_proxy) {
```

Comment:
Some description of what this function does and the expected inputs look would be really nice.

Reply:
I simply copied the comment from about 15 lines up. It's meant to cover all four of these functions, but I guess it's not so clear since this function has a different name.